priv/static/assets/js/arizona-worker.js

```diff
@@ -111,32 +111,39 @@ function handleEvent(data) {
   }
 }
 
-function applyPatch([target, changes]) {
-  const tree =
-    target === "root" ? state.tree : getTargetTree(target, state.tree);
+function applyPatch([[, ...target], changes]) {
+  const tree = getTargetTree(target, state.tree[1]);
   changes.forEach((c) => {
     applyChanges(c, tree);
   });
-  const html = tree.flat(Infinity).join("");
-  return { target, html };
+  const html = zip(tree[0], tree[1]);
+  return { target: [0, ...target], html };
+}
+
+function getTargetTree(path, tree) {
+  if (path.length === 1) {
+    return tree[path[0]];
+  } else {
+    const [i, ...rest] = path;
+    return getTargetTree(rest, tree[i]);
+  }
 }
 
 function applyChanges([indexes, v], tree) {
   if (indexes.length === 1) {
-    tree[indexes[0]] = v;
+    tree[1][indexes[0]] = v;
   } else {
     const [i, ...rest] = indexes;
-    applyChanges([rest, v], tree[i]);
+    applyChanges([rest, v], tree[1][i]);
   }
 }
 
-function getTargetTree(path, tree) {
-  if (path.length === 1) {
-    return tree[path[0]];
-  } else {
-    const [i, ...rest] = path;
-    return getTargetTree(rest, tree[i]);
+function zip(staticArr, dynamicArr) {
+  let str = "";
+  for (let i = 0; i < Math.max(staticArr.length, dynamicArr.length); i++) {
+    str += `${staticArr[i] || ""}${dynamicArr[i] || ""}`;
   }
+  return str;
```

Comment:
The same thing happens now on the server side: We "zip" static and dynamic lists.
`tree[0]` is a static list and `tree[1]` is dynamic.
I was unaware that `static` is a reserved word in Javascript, so I renamed the function args `staticArr` and `dynamicArr` instead of just `static` and `dynamic`.
> Arr = array.